### PR TITLE
secure-delete: init at 3.1

### PR DESCRIPTION
**Pull request by xdettlaff**
## Description of changes

Gutmann method based tools for securely wiping data from files, free disk space, swap and memory: srm, sfill, sswap and sdmem.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by xdettlaff** [comment]
@balsoft ping

**Review by AleXoundOS — APPROVED** [review]

File: pkgs/by-name/se/secure-delete/package.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{ stdenv, fetchFromGitHub, fetchurl, lib }:
```
Comment (AndersonTorres) [inline pkgs/by-name/se/secure-delete/package.nix:1]:
```suggestion
{ lib, stdenv, fetchurl}:

```

File: pkgs/by-name/se/secure-delete/package.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+stdenv.mkDerivation rec {
+  pname = "secure-delete";
+  version = "3.1";
+  src = fetchurl {
+    url =
+      "https://github.com/hackerschoice/THC-Archive/raw/96eeadb00075097c68bc2c6d10b0a7fbd5be8fae/Tools/secure_delete-${version}.tar.gz";
+    sha256 = "1mgqvmxwy36rxjzgd1m0k0qjc7xwjs2nwc5v7gqvvwg3vk8ldn59";
+  };
```
Comment (AndersonTorres) [inline pkgs/by-name/se/secure-delete/package.nix:9]:
```suggestion
  pname = "secure-delete";
  version = "3.1";

  src = fetchurl {
    url = "https://github.com/hackerschoice/THC-Archive/raw/96eeadb00075097c68bc2c6d10b0a7fbd5be8fae/Tools/secure_delete-${version}.tar.gz";
    hash = "sha256-qdhG0dzj8b3xO7swboWWvB8mMZighva+7NkMz3vd+NU=";
  };
```

File: pkgs/by-name/se/secure-delete/package.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  patches = [
```
Comment (AndersonTorres) [inline pkgs/by-name/se/secure-delete/package.nix:11]: Use fetchpatch instead.

File: pkgs/by-name/se/secure-delete/package.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      sha256 = "1ajs87lnqd3gvyj3bs9rvrsd1aa4vm01x5187mgcc5xp08l967a7";
+    })
+  ];
+  installPhase = ''
+    mkdir -p $out/bin
+    mkdir -p $out/share/man/man1
+
+    install -m 755 sfill smem sswap srm $out/bin
+    install -m 644 sfill.1 smem.1 sswap.1 srm.1 $out/share/man/man1
+  '';
```
Comment (AndersonTorres) [inline pkgs/by-name/se/secure-delete/package.nix:24]: Use installShellFiles instead

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by MikaelFangel** [comment]
Does it really make sense to add this as it was archived years ago? And when I skim the code it doesn't even look entirely finished and include some relatively big unfinished todo's 🤔

**Pull request closed without merging**